routes/spec.js

```diff
@@ -1129,26 +1130,25 @@ Currently the API has a rate limit of **60 requests/minute** and **50,000 reques
           },
         },
         route: () => '/skyblock/bazaar/:id',
-        func: (req, res, cb) => {
-          const itemId = req.params.id;
-          redis.get('skyblock_bazaar', (err, resp) => {
-            const ids = JSON.parse(resp) || [];
-            if (itemId && !itemId.includes(',') && !ids.includes(itemId)) {
-              return res.status(400).json({ error: 'Invalid itemId' });
+        func: async (request, response) => {
+          const itemId = request.params.id;
+          const data = await promisify(redis.get)('skyblock_bazaar');
+          const ids = JSON.parse(data) || [];
+          if (itemId && !itemId.includes(',') && !ids.includes(itemId)) {
+            return response.status(400).json({ error: 'Invalid itemId' });
+          }
+          try {
+            const bazaar = await buildBazaar();
+            if (!itemId) {
+              return response.json(bazaar);
             }
-            buildBazaar((err, bazaar) => {
-              if (err) {
-                return cb(err);
-              }
-              if (!itemId) {
-                return res.json(bazaar);
-              }
-              if (itemId.includes(',')) {
-                return res.json(filterObject(bazaar, itemId.split(',')));
-              }
-              return res.json(bazaar[itemId]);
-            });
-          });
+            if (itemId.includes(',')) {
+              return response.json(filterObject(bazaar, itemId.split(',')));
+            }
+            return response.json(bazaar[itemId]);
+          } catch (error) {
+            response.status(500).json({ error: error.message });
```

Comment:
We probably want to just call the callback here and let the error handler middleware deal with it and not show the user the error.